clang/lib/Sema/BoundsWideningAnalysis.cpp

```diff
@@ -511,15 +584,20 @@ BoundsMapTy BoundsWideningAnalysis::GetStmtOut(const CFGBlock *B,
 
   ElevatedCFGBlock *EB = BlockIt->second;
 
-  if (CurrStmt) {
-    auto Diff = BWUtil.Difference(EB->OutOfPrevStmt, EB->StmtKill[CurrStmt]);
-    auto StmtOut = BWUtil.Union(Diff, EB->StmtGen[CurrStmt]);
-    EB->OutOfPrevStmt = StmtOut;
-    return StmtOut;
+  if (!CurrStmt) {
+    EB->OutOfPrevStmt = EB->In;
+    return EB->In;
```

Comment:
Does this mean that a client has to call `GetStmtOut` with a null value for `CurrStmt` in order to initialize `EB->OutOfPrevStmt` (because it seems like `EB->OutOfPrevStmt` needs to have a valid value for this function to return sensible values). Should we be returning `BoundsMapTy()` if `CurrStmt` is null and initializing `EB->OutOfPrevStmt` to `EB->In` if `EB->OutOfPrevStmt` is null?